dashboard/test/ui/features/step_definitions/projects.rb

```diff
@@ -95,3 +95,12 @@
   code = "window.dashboard.project.__TestInterface.setSourceVersionInterval(#{seconds});"
   @browser.execute_script(code)
 end
+
+Then /^the project table contains ([\d]+) (?:row|rows)$/ do |expected_num|
+  actual_num = @browser.execute_script("return $('.ui-personal-projects-row').length;")
+  expect(actual_num).to eq(expected_num.to_i)
+end
+
+Then /^there is a project in the table named "([^"]*)"$/ do |expected_name|
+  element_exists?(expected_name)
```

Comment:
I don't think this does what I want it to. The step passes if I put in anything, not just project names that are actually in the table 😞. @davidsbailey, do you have any suggestions? I'm trying variations of the "is visible" step, but you might have a better idea.

Reply:
Generally speaking, you should be using jquery for this, so following the example in the "is visible" steps sounds right. Before you spend too many cycles waiting for UI tests to run, you should make sure that the jquery selectors that you plan to use in your UI test are doing the right thing when you run them manually from the browser's debug console. please ping me if you want to work together through any of this.